pycloudlib/gce/instance.py

```diff
@@ -0,0 +1,182 @@
+# This file is part of pycloudlib. See LICENSE file for license information.
+"""GCE instance."""
+
+from time import sleep
+
+import googleapiclient.discovery
+from googleapiclient.errors import HttpError
+
+from pycloudlib.gce.util import raise_on_error
+from pycloudlib.instance import BaseInstance
+
+
+class GceInstance(BaseInstance):
+    """GCE backed instance."""
+
+    _type = 'gce'
+
+    def __init__(self, key_pair, instance_id, project, zone, name=None):
+        """Set up the instance.
+
+        Args:
+            key_pair: A KeyPair for SSH interactions
+            instance_id: Id returned when creating the instance
+            project: Project instance was created in
+            zone: Zone instance was created in
+        """
+        if project is None or zone is None:
+            raise ValueError("kwargs 'project' and 'zone' are required. "
+                             "Project: {}, Zone: {}".format(project, zone))
+        super().__init__(key_pair)
+        self.instance_id = instance_id
+        self._name = name
+        self.project = project
+        self.zone = zone
+        self._ip = None
+        self.instance = googleapiclient.discovery.build(
+            'compute', 'v1', cache_discovery=False
+        ).instances()
+
+    def __repr__(self):
+        """Create string representation of class."""
+        return '{}(instance_id={})'.format(
+            self.__class__.__name__,
+            self.instance_id,
+        )
+
+    @property
+    def name(self):
```

Comment:
Can we add a def id(self): that is an alias to name property as current callers on ec2 typically use this alias since our cloud get_instance calls for an instance_id param (instead of name)

Reply:
actual instance_id is better thanks .